### PR TITLE
Final draft: adding Fedora 27, Fedora 28, openSUSE 42.3, and Ubuntu 18.04

### PR DESCRIPTION
**Pull request by ghost**
Please feel free to add people to the code review...

**Private drop:**
http://dotnetcli.blob.core.windows.net/jbeisner/Sdk/1.1.9-servicing-005294/

**.cliversion**
No longer needed as the 'bootstrap' step is removed from 'run-build'

**run-build.ps1 
run-build.sh**
Removing the unnecessary 'bootstrap' step
Using a temp 1.0.4-btstrp SDK for the new Linux distros

**README.md** 
Add info on the new distros

**build/Microsoft.DotNet.Cli.DependencyVersions.props** 
Update to the newer version of the runtime, el al

**build/Microsoft.DotNet.Cli.Monikers.props 
test/Microsoft.DotNet.Cli.Utils.Tests/GivenAProjectToolsCommandResolver.cs** 
1.1.8 -> 1.1.9

**build/Microsoft.DotNet.Cli.Publish.targets 
build/compile/Microsoft.DotNet.Cli.LzmaArchive.targets** 
Using clearer conditionals; fixing logic

**build_projects/dotnet-cli-build/dotnet-cli-build.csproj**
Variable was incorrect

**test/dotnet-migrate.Tests/GivenThatIWantToMigrateDeprecatedProjects.cs 
test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs 
test/dotnet-publish.Tests/GivenDotnetPublishPublishesProjects.cs** 
Disabling tests for the new Linux distros where appropriate 
Related: https://github.com/dotnet/cli/issues/9310

**scripts/obtain/dotnet-install.ps1 
scripts/obtain/dotnet-install.sh**
Using the most updated versions from CLI:master

**scripts/dockerrun.sh** 
Adding the new Linux distros
Removing access tokens and passwords so they do not appear as clear-text in logs.

### CODE REVIEW

File: scripts/dockerrun.sh [hunk 1]
```diff
@@ -138,10 +144,7 @@ docker run $INTERACTIVE -t --rm --sig-proxy=true \
     -e RELEASESUFFIX \
     -e COREFXAZURECONTAINER \
     -e AZUREACCOUNTNAME \
-    -e AZUREACCESSTOKEN \
-    -e VSOPASSWORD \
     -e RELEASETOOLSGITURL \
     -e CORESETUPBLOBROOTURL \
-    -e CORESETUPBLOBACCESSTOKEN \
```
Comment (eerhardt) [inline scripts/dockerrun.sh]: Isn't this variable still being used in Prepare.targets?  If you remove this, it won't be passed through to the docker container.

Comment (johnbeisner) [reply scripts/dockerrun.sh]: @eerhardt 
All access tokens/passwords are VSO secrets now. They cannot be passed thru via the environment  - they must be explicitly set on a command line; typically as a MSBuild property.

File: build/Microsoft.DotNet.Cli.Prepare.targets [hunk 2]
```diff
@@ -96,12 +96,13 @@
...
 
-      <ProductMonikerRid Condition=" '$(Rid)' == 'ubuntu.16.04-x64' Or 
-                                     '$(Rid)' == 'ubuntu.16.10-x64' Or 
-                                     '$(Rid)' == 'fedora.23-x64' Or 
-                                     '$(Rid)' == 'fedora.24-x64' Or 
-                                     '$(Rid)' == 'opensuse.13.2-x64' Or 
-                                     '$(Rid)' == 'opensuse.42.1-x64' ">$(Rid)</ProductMonikerRid>
+      <ProductMonikerRid Condition=" '$(Rid)' == 'ubuntu.16.04-x64' Or
+                                     '$(Rid)' == 'ubuntu.18.04-x64' Or
+                                     '$(Rid)' == 'fedora.24-x64' Or
```
Comment (janvorli) [inline build/Microsoft.DotNet.Cli.Prepare.targets]: Should Fedora 24 go away here too?

File: build/Microsoft.DotNet.Cli.Prepare.targets [hunk 3]
```diff
@@ -134,9 +135,12 @@
     </PropertyGroup>
 
     <PropertyGroup>
-      <HasAdditionalSharedFramework Condition="'$(Rid)' != 'ubuntu.16.10-x64' AND
+      <HasAdditionalSharedFramework Condition="'$(Rid)' != 'ubuntu.18.04-x64' AND
                                                '$(Rid)' != 'fedora.24-x64' AND
```
Comment (janvorli) [inline build/Microsoft.DotNet.Cli.Prepare.targets]: Should Fedora 24 go away here too?

File: build_projects/shared-build-targets-utils/Utils/Monikers.cs [hunk 4]
```diff
@@ -13,18 +13,28 @@ public static string GetBadgeMoniker()
...
+                    return "Ubuntu_x64";
                 case "ubuntu.16.04-x64":
                     return "Ubuntu_16_04_x64";
-                case "ubuntu.16.10-x64":
-                    return "Ubuntu_16_10_x64";
-                case "fedora.23-x64":
-                    return "Fedora_23_x64";
+                case "ubuntu.18.04-x64":
+                    return "Ubuntu_18_04_x64";
                 case "fedora.24-x64":
```
Comment (janvorli) [inline build_projects/shared-build-targets-utils/Utils/Monikers.cs]: Should Fedora 24 go away here too?

File: build_projects/update-dependencies/update-dependencies.csproj [hunk 5]
```diff
@@ -6,7 +6,7 @@
     <TargetFramework>netcoreapp1.0</TargetFramework>
     <AssemblyName>update-dependencies</AssemblyName>
     <OutputType>Exe</OutputType>
-    <RuntimeIdentifiers>win7-x64;win7-x86;osx.10.10-x64;osx.10.11-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;ubuntu.16.10-x64;centos.7-x64;rhel.7.2-x64;debian.8-x64;fedora.23-x64;fedora.24-x64;opensuse.13.2-x64;opensuse.42.1-x64</RuntimeIdentifiers>
+    <RuntimeIdentifiers>win7-x64;win7-x86;osx.10.10-x64;osx.10.11-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;ubuntu.18.04-x64;centos.7-x64;rhel.7.2-x64;debian.8-x64;fedora.24-x64;fedora.27-x64;fedora.28-x64;opensuse.42.1-x64;opensuse.42.3-x64</RuntimeIdentifiers>
```
Comment (janvorli) [inline build_projects/update-dependencies/update-dependencies.csproj]: Should Fedora 24 go away here too?

File: buildpipeline/pipeline.json [hunk 6]
```diff
@@ -32,17 +32,41 @@
           }
         },
         {
-          "Name": "DotNet-CLI-Fedora23-x64",
+          "Name": "DotNet-CLI-Fedora24-x64",
```
Comment (janvorli) [inline buildpipeline/pipeline.json]: Should Fedora 24 go away here too?

File: netci.groovy [hunk 7]
```diff
@@ -30,19 +45,64 @@ platformList.each { platform ->
...
-        buildCommand = "./build.sh --skip-prereqs --configuration ${configuration} --targets Default"
+        if (os == 'CentOS7.1') {
+            osUsedForMachineAffinity = 'Ubuntu16.04';
+            dockerFlag = "centos"
+        }
+        else if (os == 'Debian8.2') {
+            osUsedForMachineAffinity = 'Ubuntu16.04';
+            dockerFlag = "debian"
+        }
+        else if (os == 'Fedora24') {
```
Comment (janvorli) [inline netci.groovy]: Should Fedora 24 go away here too?

File: scripts/dockerrun.sh [hunk 8]
```diff
@@ -72,19 +75,28 @@ if [ -z "$DOCKERFILE" ]; then
             export DOCKERFILE=scripts/docker/debian
         elif [ "$(cat /etc/*-release | grep -cim1 fedora)" -eq 1 ]; then
             echo "Detected current OS as Fedora, determining fedora version to use..."
-            if [ "$(cat /etc/*-release | grep -cim1 23)" -eq 1 ]; then
-                echo "using 'fedora.23' image"
-                export DOCKERFILE=scripts/docker/fedora.23
+            if [ "$(cat /etc/*-release | grep -cim1 24)" -eq 1 ]; then
```
Comment (janvorli) [inline scripts/dockerrun.sh]: Should Fedora 24 go away here too?

File: TestAssets/DesktopTestProjects/AppWithProjTool2Fx/App.csproj [hunk 9]
```diff
@@ -3,7 +3,7 @@
     <TargetFrameworks>netcoreapp1.1;net451</TargetFrameworks>
     <OutputType>Exe</OutputType>
     <PackageTargetFallback Condition=" '$(TargetFramework)' == 'netcoreapp1.1' ">$(PackageTargetFallback);portable-net45+win8;dnxcore50</PackageTargetFallback>
-    <RuntimeIdentifiers>win7-x64;win7-x86;osx.10.10-x64;osx.10.11-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;centos.7-x64;rhel.7.2-x64;debian.8-x64;fedora.23-x64;opensuse.13.2-x64</RuntimeIdentifiers>
+    <RuntimeIdentifiers>win7-x64;win7-x86;osx.10.10-x64;osx.10.11-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;ubuntu.18.04-x64;centos.7-x64;rhel.7.2-x64;debian.8-x64;fedora.24-x64;fedora.27-x64;fedora.28-x64;opensuse.42.1-x64;opensuse.42.3-x64</RuntimeIdentifiers>
```
Comment (janvorli) [inline TestAssets/DesktopTestProjects/AppWithProjTool2Fx/App.csproj]: Should Fedora 24 go away here too?

File: TestAssets/TestPackages/PackageWithFakeNativeDep/PackageWithFakeNativeDep.csproj [hunk 10]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <TargetFramework>net45</TargetFramework>
     <AssemblyName>PackageWithFakeNativeDep</AssemblyName>
-    <RuntimeIdentifiers>win7-x64;win7-x86;osx.10.11-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;centos.7-x64;rhel.7.2-x64;debian.8-x64;fedora.23-x64;opensuse.13.2-x64;osx.10.10-x64;rhel.7-x64</RuntimeIdentifiers>
+    <RuntimeIdentifiers>win7-x64;win7-x86;osx.10.10-x64;osx.10.11-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;ubuntu.18.04-x64;centos.7-x64;rhel.7.2-x64;debian.8-x64;fedora.24-x64;fedora.27-x64;fedora.28-x64;opensuse.42.1-x64;opensuse.42.3-x64</RuntimeIdentifiers>
```
Comment (janvorli) [inline TestAssets/TestPackages/PackageWithFakeNativeDep/PackageWithFakeNativeDep.csproj]: Should Fedora 24 go away here too?

File: TestAssets/TestPackages/dotnet-hello/v1/dotnet-hello/dotnet-hello.csproj [hunk 11]
```diff
@@ -5,7 +5,7 @@
     <TargetFramework>netcoreapp1.1</TargetFramework>
     <AssemblyName>dotnet-hello</AssemblyName>
     <OutputType>Exe</OutputType>
-    <RuntimeIdentifiers>win7-x64;win7-x86;osx.10.10-x64;osx.10.11-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;centos.7-x64;rhel.7.2-x64;debian.8-x64</RuntimeIdentifiers>
+    <RuntimeIdentifiers>win7-x64;win7-x86;osx.10.10-x64;osx.10.11-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;ubuntu.18.04-x64;centos.7-x64;rhel.7.2-x64;debian.8-x64;fedora.24-x64;fedora.27-x64;fedora.28-x64;opensuse.42.1-x64;opensuse.42.3-x64</RuntimeIdentifiers>
   </PropertyGroup>
```
Comment (janvorli) [inline TestAssets/TestPackages/dotnet-hello/v1/dotnet-hello/dotnet-hello.csproj]: Should Fedora 24 go away here too?

File: TestAssets/TestPackages/dotnet-hello/v2/dotnet-hello/dotnet-hello.csproj [hunk 12]
```diff
@@ -6,7 +6,7 @@
     <TargetFramework>netcoreapp1.1</TargetFramework>
     <AssemblyName>dotnet-hello</AssemblyName>
     <OutputType>Exe</OutputType>
-    <RuntimeIdentifiers>win7-x64;win7-x86;osx.10.10-x64;osx.10.11-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;centos.7-x64;rhel.7.2-x64;debian.8-x64</RuntimeIdentifiers>
+    <RuntimeIdentifiers>win7-x64;win7-x86;osx.10.10-x64;osx.10.11-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;ubuntu.18.04-x64;centos.7-x64;rhel.7.2-x64;debian.8-x64;fedora.24-x64;fedora.27-x64;fedora.28-x64;opensuse.42.1-x64;opensuse.42.3-x64</RuntimeIdentifiers>
```
Comment (janvorli) [inline TestAssets/TestPackages/dotnet-hello/v2/dotnet-hello/dotnet-hello.csproj]: Should Fedora 24 go away here too?

File: TestAssets/TestProjects/MSBuildTestApp/MSBuildTestApp.csproj [hunk 13]
```diff
@@ -2,8 +2,9 @@
   <PropertyGroup>
     <OutputType>Exe</OutputType>
     <TargetFramework>netcoreapp1.1</TargetFramework>
-    <RuntimeIdentifiers>win7-x64;win7-x86;osx.10.10-x64;osx.10.11-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;ubuntu.16.10-x64;centos.7-x64;rhel.7.2-x64;debian.8-x64;fedora.23-x64;fedora.24-x64;opensuse.13.2-x64;opensuse.42.1-x64</RuntimeIdentifiers>
+    <RuntimeIdentifiers>win7-x64;win7-x86;osx.10.10-x64;osx.10.11-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;ubuntu.18.04-x64;centos.7-x64;rhel.7.2-x64;debian.8-x64;fedora.24-x64;fedora.27-x64;fedora.28-x64;opensuse.42.1-x64;opensuse.42.3-x64</RuntimeIdentifiers>
```
Comment (janvorli) [inline TestAssets/TestProjects/MSBuildTestApp/MSBuildTestApp.csproj]: Should Fedora 24 go away here too?

File: build/Microsoft.DotNet.Cli.Prepare.targets (hunk 2 above)
Comment (johnbeisner) [reply build/Microsoft.DotNet.Cli.Prepare.targets]: @janvorli 
I would like to keep the ability to build Fedora 24 and openSUSE 42.1 - we will not build either officially. Moreover, if we remove Fedora 24, CI will most likely break.

Comment (janvorli) [reply build/Microsoft.DotNet.Cli.Prepare.targets]: Ah, sorry for the misunderstanding then. I've thought that it was to be removed completely.

File: build/Microsoft.DotNet.Cli.Publish.targets [hunk 14]
```diff
@@ -7,7 +7,7 @@
 
   <!-- PUBLISH_TO_AZURE_BLOB env variable set by CI -->
   <Target Name="Publish"
-          Condition=" '$(PUBLISH_TO_AZURE_BLOB)' != '' "
+          Condition=" '$(PUBLISH_TO_AZURE_BLOB)' == 'true' "
```
Comment (wli3) [inline build/Microsoft.DotNet.Cli.Publish.targets]: deliberate? bool in msubild is not used consistently

Comment (johnbeisner) [reply build/Microsoft.DotNet.Cli.Publish.targets]: This was indeed a deliberate change.

File: buildpipeline/pipeline.json [hunk 15]
```diff
@@ -32,17 +32,41 @@
...
           "ReportingParameters": {
-            "OperatingSystem": "Fedora 23",
+            "OperatingSystem": "Fedora 24",
             "Type": "build/product/",
             "Platform": "x64"
           }
         },
         {
-          "Name": "DotNet-CLI-opensuse13.2-x64",
```
Comment (wli3) [inline buildpipeline/pipeline.json]: replacement of opensuse13.2 is fedora?

File: tools/Archiver/Archiver.csproj [hunk 16]
```diff
@@ -4,7 +4,7 @@
   <PropertyGroup>
     <TargetFramework>netcoreapp1.1</TargetFramework>
     <OutputType>Exe</OutputType>
-    <RuntimeIdentifiers>win7-x64;win7-x86;osx.10.11-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;centos.7-x64;rhel.7.2-x64;debian.8-x64;fedora.24-x64;opensuse.42.1-x64;osx.10.10-x64;rhel.7-x64</RuntimeIdentifiers>
+    <RuntimeIdentifiers>win7-x64;win7-x86;osx.10.10-x64;osx.10.11-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;ubuntu.18.04-x64;centos.7-x64;rhel.7.2-x64;debian.8-x64;fedora.24-x64;fedora.27-x64;fedora.28-x64;opensuse.42.1-x64;opensuse.42.3-x64</RuntimeIdentifiers>
```
Comment (wli3) [inline tools/Archiver/Archiver.csproj]: The latest osx is 10.13, is that ok?

File: scripts/obtain/dotnet-install.ps1 [hunk 17]
```diff
@@ -10,18 +10,22 @@
     Installs dotnet cli. If dotnet installation already exists in the given directory
     it will update it only if the requested version differs from the one already installed.
 .PARAMETER Channel
-    Default: preview
-    Channel is the way of reasoning about stability and quality of dotnet. This parameter takes one of the values:
-    - future - Possibly unstable, frequently changing, may contain new finished and unfinished features
-    - preview - Pre-release stable with known issues and feature gaps
-    - production - Most stable releases
```
Comment (wli3) [inline scripts/obtain/dotnet-install.ps1]: All installer script change is just a post of master right?

**Comment by wli3** [comment]
And any process on https://github.com/dotnet/cli/issues/9310?

**Comment by wli3** [comment]
I read all. Other than comments above it is good

**Comment by johnbeisner** [comment]
@dotnet/dotnet-cli 
Is this one ready to merge?

**Comment by johnbeisner** [comment]
@dotnet/dotnet-cli
Is this one ready to merge?

File: scripts/obtain/dotnet-install.ps1 (hunk 17 above)
Comment (johnbeisner) [reply scripts/obtain/dotnet-install.ps1]: Yes, just a copy of the scripts in master.

File: tools/Archiver/Archiver.csproj (hunk 16 above)
Comment (johnbeisner) [reply tools/Archiver/Archiver.csproj]: We are still okay here.

File: buildpipeline/pipeline.json (hunk 15 above)
Comment (johnbeisner) [reply buildpipeline/pipeline.json]: This is not a replacement - it just looks like it via GitHub's diff.

**Review by wli3 — APPROVED** [review]

**Comment by johnbeisner** [comment]
@dotnet-bot test this please

**Pull request merged**